shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -289,6 +289,8 @@ - (void)setViewController:(FlutterViewController*)viewController {
 
 - (void)attachView {
   self.iosPlatformView->attachView();
+  _textInputPlugin.get().viewController = [self viewController];
```

Comment:
what if the view controller changes during the lifetime of the flutterengine?
also flutter holdsa weak ptr to the viewcontroller, I am not sure what will happen to the assign property of _textInputPlugin if the pointer is nulled in flutterengine, cc @gaaclarke

Reply:
@chunhtai I also set this property below on like 476, I'm not sure whether or not that would fix the issue – how would I test that? I don't know when the view controller would change

Reply:
I think the viewcontroller of the textinputplugin need to be set in FlutterEngine.setViewController, and it also need to nil it in FlutterEngine.notifyViewControllerDeallocated. To test this, you can swap out view controller when you run the FlutterEngine and see if things still work

Reply:
@chunhtai I may need some more guidance on how to test this...I tried the following, but the notifyViewControllerDeallocated method doesn't seem to get hit when swapping out the view controller – attachView does get called again, so I think it is "working":

```
@UIApplicationMain
@objc class AppDelegate: FlutterAppDelegate {
  override func application(
    _ application: UIApplication,
    didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
  ) -> Bool {
    
    var controller = window.rootViewController as! FlutterViewController
    
    let criticChannel = FlutterMethodChannel(name: "test.viewController", binaryMessenger: controller as! FlutterBinaryMessenger)
    criticChannel.setMethodCallHandler { (call: FlutterMethodCall, result: @escaping FlutterResult) -> Void in
        if call.method == "swap" {
            let engine = controller.engine!
            engine.viewController = nil
            self.window.rootViewController = FlutterViewController(engine: engine, nibName: controller.nibName, bundle: controller.nibBundle)
            controller = self.window.rootViewController as! FlutterViewController
            result(nil)
        } else {
            result(FlutterMethodNotImplemented)
        }
    }
    
    GeneratedPluginRegistrant.register(with: self)
    return super.application(application, didFinishLaunchingWithOptions: launchOptions)
  }
}
```

Reply:
@chunhtai I updated the `setupIndirectScribbleInteraction` method to take in the `FlutterViewController` instead of setting it directly in the engine, that way we're setting up that interaction every time the view controller changes. I also added a resetViewController method to nil the viewController from that `notifyViewControllerDeallocated` method, though I still haven't managed to actually have that get hit 🤷‍♂️

Reply:
you will need to hold a strong reference FlutterViewController somewhere and pass that into the flutter engine. and then nil the strong reference, this should cause garbage collection to dealloc the FlutterViewController, which will then call the notifyViewControllerDeallocated.

can you write a unit test for this? so that we don't break this unintentionally.